### ISSUE TITLE
Unable to run in on localserver

### ISSUE THREAD

**Issue by shashvattrip** [maximilianh/ucscBeacon#1]
Hey, 
I have ubuntu 14.04. I git cloned your repo it in /var/www/html
I then tried out your commands but it just outputs the file contents to stdin. Please help me out.
Thanks!

**Comment by maximilianh** [comment]
The instructions do not mention /var/www/html but your cgi-bin directory. Please google it again or look at the readme, I'm on the road it might be something like usr/lib/cgi-bin

**Comment by maximilianh** [comment]
Yes it's /usr/lib/cgi-bin on ubuntu, that's written in the readme file I
think. I'll make this clearer.

Just move everything into /usr/lib/cgi-bin and it should work.

cheers
Max

**Comment by shashvattrip** [comment]
Hey
I did try it first in /usr/lib/cgi-bin but then got this error

Resolving localhost (localhost)... 127.0.0.1
Connecting to localhost (localhost)|127.0.0.1|:80... connected.
HTTP request sent, awaiting response... 404 Not Found
2015-03-07 00:27:24 ERROR 404: Not Found.

Which makes sense because all the server files are usually set up in /var/www/html for ubuntu 12.04 which is why I moved it there and then tried it out.
And thanks for helping me out. I really appreciate it! :D

**Comment by maximilianh** [comment]
what exactly did you try? Did you try localhost:/cgi-bin/ucscBeacon/query ?

/var/www/html is only for html files.
/usr/lib/cgi-bin is for scripts.

this is not an html file but a script so it really goes into
/usr/lib/cgi-bin.

I can try this on a fresh ubuntu install tomorrow.

**Comment by shashvattrip** [comment]
Yes, I did try 
http://localhost/cgi-bin/ucscBeacon/query
But it just gave me this error, 
The requested URL /cgi-bin/ucscBeacon/query was not found on this server.

Here's what I have done till now, in the following order, 
cd /usr/lib/cgi-bin 
git clone https://github.com/maximilianh/ucscBeacon.git

sudo apachctl -k restart  //this gives me an error that apachctl : bad command, so I do instead
sudo apache2ctl -k restart

Next I try to test it by typing in the terminal, this is what I get :

/usr/lib/cgi-bin/ucscBeacon(branch:master) » wget 'http://localhost/cgi-bin/ucscBeacon/query' -O -

--2015-03-07 01:06:09--  http://localhost/cgi-bin/ucscBeacon/query
Resolving localhost (localhost)... 127.0.0.1
Connecting to localhost (localhost)|127.0.0.1|:80... connected.
HTTP request sent, awaiting response... 404 Not Found
2015-03-07 01:06:09 ERROR 404: Not Found.

**Comment by maximilianh** [comment]
OK I've figured it out, one needs to activate the cgi-bin module first in
Ubuntu, looks like I did that a long time ago on mine, I just tried on a
fresh ubuntu install, these are the commands (now in the README file too):

sudo apt-get install apache2 git
sudo a2enmod cgi
sudo service apache2 restart
cd /usr/lib/cgi-bin
git clone https://github.com/maximilianh/ucscBeacon.git

**Comment by shashvattrip** [comment]
Ah. Perfect! It works now.
Thanks a ton!

**Comment by maximilianh** [comment]
can you tell me which data types you want to serve? and maybe a little bit
of background (institution?).

are you planning to serve allele frequencies? How "clinical" is the dataset?

**Comment by shashvattrip** [comment]
The goal of the project is to implement a JavaScript frontend for configuring and monitoring an implementation of the beacon API. The tool should allow people without software development background to easily make a reference beacon implementation serve the data they want to share. 
I want to do this for multiple beacons and am going to begin with yours. 
I am doing this with Global Alliance for Genomics and Health organization.

And to be honest, I have no idea what data type I want to serve, I have no background in genetics, and have no idea how clinical the dataset it. :)
Check this out : Last Project : Beacon management console
https://docs.google.com/document/d/1PHPDTzD4Z6xVoxN0Q1F8yEPclSri-JA1vLvMTrLYUi8/edit#

**Comment by maximilianh** [comment]
oh I see. sounds cool! Who are you working with on this?

**Comment by shashvattrip** [comment]
Working with @mcupak https://github.com/mcupak
